### PR TITLE
feat(FR-2806): auto-fill login API endpoint from VITE_DEFAULT_API_ENDPOINT in dev

### LINKED ISSUE #7234: Auto-fill login API endpoint from VITE_DEFAULT_API_ENDPOINT in dev
## Problem

When developing against a per-PR backend test server, the API endpoint shown on the WebUI login screen still points to whatever was last used (from `localStorage` or a previously connected client). Developers manually re-type the target endpoint on every cold start, and — when a session is still live for a **different** endpoint — the orchestrator may silently re-login to the wrong server.

## Proposal

Add a Vite-only env var that the dev-server skill auto-derives from the current PR description:

- `VITE_DEFAULT_API_ENDPOINT` (effective only in `import.meta.env.DEV`)
- When set, the LoginView's `apiEndpoint` initial state and the form field are pre-filled with this value, **overriding the localStorage-stored endpoint**.
- If the previously connected client targets a **different** endpoint, the login orchestration force-logs-out that session and shows the login panel pre-populated with the env-var endpoint, instead of attempting a silent re-login.

## Dev-server skill rule

The `dev-server` skill (`.claude/skills/dev-server/SKILL.md`) gains a new step that scans the current branch's PR description for a backend address:

- Bare IP (e.g. `10.0.1.5`) → `http://10.0.1.5:8090`
- Bare `host:port` (e.g. `10.0.1.5:9090`) → `http://10.0.1.5:9090`
- Full URL (`http://...` / `https://...`) → use as-is
- Otherwise → omit `VITE_DEFAULT_API_ENDPOINT`

User-set env or explicit prompt mentions take precedence over PR-description detection (consistent with the existing `VITE_THEME_HEADER_COLOR` / `PORTLESS_APP_NAME` rules).

## Acceptance criteria

- `VITE_DEFAULT_API_ENDPOINT=http://1.2.3.4:8090 pnpm dev` → cold-start login form shows `http://1.2.3.4:8090` in the endpoint field.
- A previously stored endpoint in `localStorage.backendaiwebui.api_endpoint` does not override the env-var value in dev mode.
- If a session for a different endpoint is active when the page loads, the orchestrator logs it out and surfaces the login panel — no silent re-login to the stale endpoint.
- In production builds (`import.meta.env.DEV === false`), the env var is ignored entirely; current behavior is unchanged.
- The dev-server skill, when invoked from a branch with a PR whose description contains an IP, prefixes `pnpm dev` with `VITE_DEFAULT_API_ENDPOINT='http://<ip>:8090'`.

JIRA Issue: FR-2806

### PR DESCRIPTION
**Pull request by yomybaby**
Resolves #7234(FR-2806)

## Summary

Add a Vite-only env var `VITE_DEFAULT_API_ENDPOINT` that pre-fills the WebUI login screen's API endpoint and force-logs out any in-memory client targeting a different backend, and update the `dev-server` skill to derive the value from the current branch's PR description automatically.

## Why

When developing against per-PR backend test servers, the API endpoint shown on the login screen still points to whatever was last used (from `localStorage` or a previously connected client). Developers manually re-type the target endpoint on every cold start, and a stale session for a different endpoint can silently silent-re-login to the wrong server. This makes "open the dev URL → land on the right backend" the default path instead of a per-session ritual.

## Changes

- `react/src/components/LoginView.tsx`
  - Read `import.meta.env.VITE_DEFAULT_API_ENDPOINT` (only when `import.meta.env.DEV`) into `devApiEndpointOverride`.
  - Initial `apiEndpoint` state uses the override over `localStorage.backendaiwebui.api_endpoint` and persists the override back into `localStorage` so other modules see it.
  - When `config.toml` later loads, the override still wins over `loginConfig.api_endpoint`, and the endpoint field is shown as editable so the developer can swap it on the fly.
- `react/src/hooks/useLoginOrchestration.ts`
  - New `enforcedEndpoint` option. When set and the in-memory client's `_config.endpoint` differs, the orchestrator logs the stale session out, nulls the global client, and opens the login panel instead of silent-re-logging in.
  - Production builds pass `undefined` (no behavior change).
- `.claude/skills/dev-server/SKILL.md`
  - New "2c. Decide the default API endpoint" section: priority order (user mention → PR body via `gh pr view --json body` → omit), the bare-IP-defaults-to-`:8090` rule, hostname / `host:port` / full-URL handling, and the regex used to scan the PR body.
  - Step 3 (compose) and step 6 (edge cases) updated to include `VITE_DEFAULT_API_ENDPOINT`.

## Test plan

- [ ] `VITE_DEFAULT_API_ENDPOINT=http://1.2.3.4:8090 pnpm dev` → cold-start login form is pre-filled with `http://1.2.3.4:8090`; endpoint field is editable.
- [ ] With a stale `localStorage.backendaiwebui.api_endpoint = "http://OLD:8090"`, the env-var value still wins on first render.
- [ ] With an in-memory `globalThis.backendaiclient` connected to a different endpoint (HMR-preserved), the orchestrator logs it out and shows the login panel — no silent re-login.
- [ ] `pnpm build` then serve → env var has no effect (`import.meta.env.DEV === false`).
- [ ] From a branch with an open PR whose body contains a bare IP, invoking the dev-server skill prefixes `pnpm dev` with `VITE_DEFAULT_API_ENDPOINT='http://<ip>:8090'`. From a branch with no PR or no IP in the body, the variable is omitted.
- [ ] `bash scripts/verify.sh` — Relay/Lint/Format pass; the only TypeScript failures are the pre-existing ones on `main` (`packages/backend.ai-client/src/client.ts`, `src/components/VFolderDeployModal.tsx`).

### CODE REVIEW

**Comment by yomybaby** [comment]
* **#7237** <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/7237?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/lablup/backend.ai-webui/7237?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>flow:merge-queue</kbd> - adds this PR to the back of the merge queue
- <kbd>flow:hotfix</kbd> - for urgent changes, fast-track this PR to the front of the merge queue

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/lablup?ref=merge-queue-instructions-in-stack-comment&prId=7237).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: react/src/hooks/useLoginOrchestration.ts [hunk 1]
```diff
@@ -134,13 +159,37 @@ export function useLoginOrchestration({
...
     if (isAppLauncherPage()) return;
 
+    // Dev-only: when an enforced endpoint is supplied (from
+    // VITE_DEFAULT_API_ENDPOINT) and the in-memory client targets a
+    // different backend, log it out and surface the login panel rather
+    // than letting the orchestrator resume the stale session.
+    if (enforcedEndpoint && isClientConnected()) {
+      const connectedEp = getConnectedEndpoint();
+      if (connectedEp !== enforcedEndpoint) {
+        // Consume any pending intentional-logout flag so the next
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply react/src/hooks/useLoginOrchestration.ts:182]: Good catch — fixed in 9e5c4e0. The hook now normalizes `enforcedEndpoint` (trim + strip trailing `/`) at the top of `useLoginOrchestration` and compares the normalized form against `getConnectedEndpoint()`. Added a regression test (`normalizes the enforced endpoint too (trailing slash and whitespace)`) covering `'  http://NEW:8090/  '` vs `'http://NEW:8090'`.

File: react/src/hooks/useLoginOrchestration.ts [hunk 2]
```diff
@@ -134,13 +159,37 @@ export function useLoginOrchestration({
...
+        // orchestration pass doesn't redundantly short-circuit on it.
+        sessionStorage.removeItem(INTENTIONAL_LOGOUT_FLAG);
+        try {
+          await onLogoutSession();
+        } catch {
+          // best-effort; the stale client may already be unreachable
+        }
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply react/src/hooks/useLoginOrchestration.ts:194]: Fixed in 9e5c4e0. The catch now logs via `useBAILogger().logger.warn` with the error before continuing to clear the global client and open the panel, so dev-only enforced-endpoint behavior stays debuggable.

File: .claude/skills/dev-server/SKILL.md [hunk 3]
```diff
@@ -88,6 +91,65 @@ Do not invent additional names or alternate hex values. If the user's `/color` a
...
+**Scanning the PR body** — do this in two passes, top-to-bottom, taking the **first** valid candidate:
+
+**Pass 1 (preferred): contextual match.** Scan only lines that look like they're naming a backend, i.e. the line contains one of these markers (case-insensitive): `test server`, `test backend`, `manager`, `api endpoint`, `endpoint`, `target server`, `dev server` (when adjacent to an address), or the line lives under a heading whose text contains `Backend`, `Test`, `Server`, or `Endpoint`. On those lines, run the address regex below.
+
+**Pass 2 (fallback): full-body match.** Only if Pass 1 found nothing, run the same regex against the whole body.
+
+**Address regex** (apply per pass):
+
+```
+(https?:\/\/)?(\b(?:25[0-5]|2[0-4]\d|[01]?\d?\d)(?:\.(?:25[0-5]|2[0-4]\d|[01]?\d?\d)){3}\b|[a-z0-9-]+(?:\.[a-z0-9-]+)+\.[a-z]{2,})(?::(\d{1,5}))?(?:\/[^\s)]*)?
```
[the remark this answers is not in the document]
Comment (yomybaby) [reply .claude/skills/dev-server/SKILL.md]: Fixed in 9e5c4e0. Changed the hostname alternative from `[a-z0-9-]+(?:\.[a-z0-9-]+)+\.[a-z]{2,}` to `[a-z0-9-]+(?:\.[a-z0-9-]+)*\.[a-z]{2,}` so 2-label hosts like `example.com` / `manager.com` are now accepted while longer ones like `api.staging.example.com` still match. Documented the change in the SKILL.md note under the regex.
